File: app/src/main/java/ru/iandreyshev/parserrss/presentation/view/IFeedPageView.java
package ru.iandreyshev.parserrss.presentation.view;

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

import ru.iandreyshev.parserrss.models.rss.ViewArticle;

public interface IFeedPageView extends IBaseView {
    @StateStrategyType(AddToEndSingleStrategy.class)
    void startUpdate(boolean isStart);

    @StateStrategyType(AddToEndSingleStrategy.class)
    void setArticles(final List<ViewArticle> newArticles);

    @StateStrategyType(SkipStrategy.class)
    void updateImages(boolean isWithoutQueue);
}

File: app/src/main/java/ru/iandreyshev/parserrss/presentation/view/IFeedPageView.kt
package ru.iandreyshev.parserrss.presentation.view

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.SkipStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

import ru.iandreyshev.parserrss.models.rss.ViewArticle

interface IFeedPageView : IBaseView {
    @StateStrategyType(AddToEndSingleStrategy::class)
    fun startUpdate(isStart: Boolean)

    @StateStrategyType(AddToEndSingleStrategy::class)
    fun setArticles(newArticles: List<ViewArticle>)

    @StateStrategyType(SkipStrategy::class)
    fun updateImages(isWithoutQueue: Boolean)
}
